### PR TITLE
CPDRP-443 notification banners for opted-out schools

### PR DESCRIPTION
**Pull request by tonyheadford**
### Context
[Jira](https://dfedigital.atlassian.net/jira/software/projects/CPDRP/boards/102?selectedIssue=CPDRP-443)
Show notification banners to schools that have chosen to opt-out (diy or no ECTs) if they have been recruited by a provider

### Changes proposed in this pull request
Change  condition under which the `SchoolRecruitedTransitionComponent` decides whether to render or not
Change `PartnershipActivationJob` to clear the `opt_out_of_updates` flag on `SchoolCohort`

### Guidance to review
Given a school that has chosen either the DIY or no ECTs programme choice for the current cohort:
1. create a partnership for the current cohort
2. Observe that during the challenge window, when you visit the schools dashboard and select the opted out cohort you should see the banner warning the school of the recruitment and providing the ability to challenge it.

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### CODE REVIEW

File: app/components/school_recruited_transition_component.rb [hunk 1]
```diff
@@ -6,8 +6,8 @@ def initialize(school_cohort:)
   end
 
   def render?
-    return false unless school_cohort.school_chose_cip?
     return false if partnership.blank?
+    return false unless could_show_banner_for_programme_choice?
 
     partnership.in_challenge_window? && !partnership.challenged?
```
[the remark this answers is not in the document]
Comment (ethanmills) [reply app/components/school_recruited_transition_component.rb:12]: This and the pending flag were developed in parallel

[the remark this answers is not in the document]
Comment (tonyheadford) [reply app/components/school_recruited_transition_component.rb:12]: I'm not sure what the purpose of the `pending` flag is in the process... should it remain `true` if a partnership is challenged or should challenging a partnership revert the whole partnership to a different state?

[the remark this answers is not in the document]
Comment (tonyheadford) [reply app/components/school_recruited_transition_component.rb:12]: I think what we currently have is fine as the `pending` flag seems to be slightly different i.e. partnership can be pending and challenged if I have understood correctly.

[the remark this answers is not in the document]
Comment (ethanmills) [reply app/components/school_recruited_transition_component.rb:12]: That wasn't the intention of the flag, but if that's the implementation then that's fine. Happy with this

**Review by ethanmills — APPROVED** [review]
Sorry, looks good!

**Pull request merged**